### PR TITLE
feat: events can be stored in a mongo database

### PR DESCRIPTION
**Pull request by rehia**

### CODE REVIEW

File: src/main/kotlin/io/saagie/whyat/config/EventDaoConfiguration.kt [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        private val environment: Environment) {
+
+    @Value("\${whyat.event.dao}")
+    var daoName = "mongo"
+
+    @Bean
+    fun eventDao(): EventDao {
+        val dao = when (daoName) {
+            "", "mongo", "mongodb" -> MongoDao(context.getBean(Jongo::class.java))
+            "hdfs.avro", "avro" -> {
```
Comment (glours) [inline src/main/kotlin/io/saagie/whyat/config/EventDaoConfiguration.kt]: Extract each dao initialization in private function ?

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -2,4 +2,10 @@ hdfs.url=${WHYAT_HDFS_URL}
 hdfs.path=${WHYAT_PATH}
 server.port=${WHYAT_PORT:8080}
 avro.maxBatchSize=${AVRO_MAX_BATCH_SIZE:1000}
-avro.batchIdle=${AVRO_BATCH_IDLE:2000}
\ No newline at end of file
+avro.batchIdle=${AVRO_BATCH_IDLE:2000}
+spring.data.mongodb.database=${WHYAT_MONGO_DB:whyat}
+spring.data.mongodb.host=${MONGO_IP:127.0.0.1}
+spring.data.mongodb.port=${MONGO_PORT:27020}
+spring.data.mongodb.username=${WHYAT_MONGO_USER:whyat}
```
Comment (glours) [inline src/main/resources/application.properties]: remove default value for user and password please

Comment (rehia) [reply src/main/resources/application.properties]: sure ? could be used by default to create an empty db :p

File: src/main/kotlin/io/saagie/whyat/config/EventDaoConfiguration.kt (hunk 1 above)
Comment (rehia) [reply src/main/kotlin/io/saagie/whyat/config/EventDaoConfiguration.kt]: clearly violates OCP...
Probably even need an abstract factory !
But let's go for method extraction for the moment

File: src/main/resources/application.properties (hunk 2 above)
Comment (glours) [reply src/main/resources/application.properties]: if your db doesn't exist, your default user too :-p

File: src/main/kotlin/io/saagie/whyat/config/EventDaoConfiguration.kt (hunk 1 above)
Comment (glours) [reply src/main/kotlin/io/saagie/whyat/config/EventDaoConfiguration.kt]: giphy 1

**Review by glours — APPROVED** [review]

**Pull request merged**